### PR TITLE
VRVU-56: Added API for fetching patient identifier values

### PR DESCRIPTION
**Pull request by druchniewicz**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/biometric/api/contract/IdentifierResponse.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public class IdentifierResponse {
+
+  private String participantUuid;
+
+  private String identifierTypeName;
+
+  private String identifierValue;
+
+  public IdentifierResponse(
+      String participantUuid, String identifierTypeName, String identifierValue) {
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/module/biometric/api/contract/IdentifierResponse.java:22]: Thanks @druchniewicz  , There is minor indentation to fix. keeping everything on one single line

File: api/src/main/java/org/openmrs/module/biometric/api/service/ParticipantService.java [hunk 2]
```diff
@@ -143,4 +141,11 @@ List<PatientResponse> findPatientsByUuids(Set<String> uuids)
    */
   List<SyncTemplateResponse> getBiometricDataByParticipantIds(Set<String> uuids);
 
+  /**
+   * Retrieves all patient's identifiers of given type
+   *
+   * @param identifierName identifier type name
+   * @return
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/module/biometric/api/service/ParticipantService.java:148]: Returning is missing a comment here

Comment (druchniewicz) [reply api/src/main/java/org/openmrs/module/biometric/api/service/ParticipantService.java]: right, corrected

File: api/src/main/java/org/openmrs/module/biometric/api/contract/IdentifierResponse.java (hunk 1 above)
Comment (druchniewicz) [reply api/src/main/java/org/openmrs/module/biometric/api/contract/IdentifierResponse.java:21]: this is automatic formatting made by plugin that we use in CfL (google-java-code formatter)

File: api/src/main/java/org/openmrs/module/biometric/api/service/ParticipantService.java [hunk 3]
```diff
@@ -143,4 +141,11 @@ List<PatientResponse> findPatientsByUuids(Set<String> uuids)
    */
   List<SyncTemplateResponse> getBiometricDataByParticipantIds(Set<String> uuids);
 
+  /**
+   * Retrieves all patient's identifiers of given type
+   *
+   * @param identifierName identifier type name
+   * @return list of patient identifiers
+   */
+  List<PatientIdentifier> getAllIdentifiersByName(String identifierName);
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/module/biometric/api/service/ParticipantService.java:150]: LGTM Thanks @druchniewicz . How about making the title as _**Fetch All patient identifier values**_   .

**Review by sherrif10 — APPROVED** [review]
well done.

File: omod/src/main/java/org/openmrs/module/biometric/web/controller/ParticipantController.java [hunk 4]
```diff
@@ -537,47 +545,72 @@ public Object retrievePersonImage(
...
-    return participantService
-        .findImagesByUuids(SanitizeUtil.sanitizeStringList(map.get(PARTICIPANT_UUIDS)));
+    return participantService.findImagesByUuids(
+        SanitizeUtil.sanitizeStringList(map.get(PARTICIPANT_UUIDS)));
+  }
+
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  @RequestMapping(value = "/identifiers/{name}", method = RequestMethod.GET)
+  public List<IdentifierResponse> getAllIdentifiersByType(
```
[the remark this answers is not in the document]
Comment (druchniewicz) [reply omod/src/main/java/org/openmrs/module/biometric/web/controller/ParticipantController.java:593]: right, I unified method names

**Review by sherrif10 — APPROVED** [review]
Thanks @druchniewicz

**Pull request merged**